internal/integrationtest/profiles/profiles_test.go

```diff
@@ -150,3 +150,257 @@ func TestCompileWithDefaultProfile(t *testing.T) {
 		jsonOut.Query(".builder_result.build_properties").MustContain(`[ "build.fqbn=arduino:avr:nano" ]`)
 	}
 }
+
+func TestInitProfile(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	// Init the environment explicitly
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+
+	_, _, err = cli.Run("sketch", "new", cli.SketchbookDir().Join("Simple").String())
+	require.NoError(t, err)
+
+	_, _, err = cli.Run("core", "install", "arduino:avr")
+	require.NoError(t, err)
+
+	integrationtest.CLISubtests{
+		{"NoProfile", initNoProfile},
+		{"ProfileCorrectFQBN", initWithCorrectFqbn},
+		{"ProfileWrongFQBN", initWithWrongFqbn},
+		{"ProfileMissingFQBN", initMissingFqbn},
+		{"ExistingProfile", initExistingProfile},
+		{"SetDefaultProfile", initSetDefaultProfile},
+	}.Run(t, env, cli)
+}
+
+func initNoProfile(t *testing.T, env *integrationtest.Environment, cli *integrationtest.ArduinoCLI) {
+	projectFile := cli.SketchbookDir().Join("Simple", "sketch.yaml")
+	// Create an empty project file
+	stdout, _, err := cli.Run("profile", "init", cli.SketchbookDir().Join("Simple").String())
+	require.NoError(t, err)
+	require.Contains(t, string(stdout), "Project file created in: "+projectFile.String())
+	require.FileExists(t, projectFile.String())
+	fileContent, err := projectFile.ReadFile()
+	require.NoError(t, err)
+	require.Equal(t, "profiles:\n", string(fileContent))
+}
+
+func initWithCorrectFqbn(t *testing.T, env *integrationtest.Environment, cli *integrationtest.ArduinoCLI) {
+	projectFile := cli.SketchbookDir().Join("Simple", "sketch.yaml")
+	// Add a profile with a correct FQBN
+	_, _, err := cli.Run("profile", "init", cli.SketchbookDir().Join("Simple").String(), "-m", "Uno", "-b", "arduino:avr:uno")
+	require.NoError(t, err)
+	require.FileExists(t, projectFile.String())
+	fileContent, err := projectFile.ReadFile()
+	require.NoError(t, err)
+	require.Equal(t, "profiles:\n  Uno:\n    fqbn: arduino:avr:uno\n    platforms:\n      - platform: arduino:avr (1.8.6)\n    libraries:\n\ndefault_profile: Uno\n", string(fileContent))
```

Comment:
`libraries:\n`
this is not a valid yaml it should be `libraries:[]\n`

In the Review comment you can find where to fix it